### DISCUSSION TITLE
Cannot connect to Supramatic 4

### DISCUSSION THREAD

**Discussion by BradleyFord** [Gifford47/HCPBridgeMqtt#64] [Q&A]
**Describe the bug**
I had my device connected a long time ago, but then removed whilst I was getting it serviced by the official team.
Just now I am putting back in, but also updated to the latest Dev_pref branch

Note: my tx/rx are flipped so I adjusted this via the UI

When going through the bus scan process, I can see it come alive and also the website being active.
But soon after the BS finishes the device turns off, which seems like the BS failed to find anything.

**To Reproduce**
1. Which motor do you have? Supramatic 4
2. Which branch/release did you use? Dev_pref
3. Which peripherals (bme280, HCsr04, ...) did you use? None

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by Gifford47** [comment]
Hi!
Please change tx/rx in both combinations to avoid any mistakes.
Then please do a full flash erase and upload the binary again.
The motor also saves some bus data. So do a reset on the motor if nothing helped until now.

**Comment by BradleyFord** [comment]
I tried both combinations just in case, not sure if I tried the erase so
will give that a try.

**Comment by BradleyFord** [comment]
I just tried after doing an erase and still getting the same results.

Is there any way I can get any logs?

**Comment by Gifford47** [comment]
Are there anys results? can i close the issue?

**Comment by BradleyFord** [comment]
> Are there anys results? can i close the issue?

I have tested it soon after this message, but no change in the results.

I need to get some time to check over the hardware again fully; but work has been crazy the last month

**Reply by acpegg** [reply]
@BradleyFord - did you find time to look over the hardware again? Did you get it working?
As you can see in my other response, my Tyson board just switches off after the BS as well, with a result 0, not 1.

**Comment by acpegg** [comment]
Hi,
I have one of Tyson's v1.3 boards with the standard supplied firmware.
I also have a Supramatic e4 motor, manufactured date 2023-06.
Every time I do a bus scan, I receive response 0.
The LED lights display on the board correctly during the bus scan.
Any known issues with the e4 motor and these boards, perhaps a specific revision that has an issue?

Thanks,
Andy

**Comment by 14yannick** [comment]
Try to Power the board by usb prior to run the busscan.    This way the board is ready when busscan starts.

**Comment by acpegg** [comment]
Thanks for suggestion. I just tried that. Still get a 0 ☹️

**Comment by TheHexaMaster** [comment]
For everyone struggling with this issue (probably ESP32-WROOM board) - there is probability you have wrong RX/TX pinout settings. If you look at the connection circuit scheme, there are RX2 / TX2 pins connected to RS485 converter, that represent GPIO16 and 17. Some aliexpress boards does not have marked pins 16 and 17 as RX2 TX2, just as 16 17, so if you use default RX0 TX0 pins on board, these pins are GPIO 1 and 3. To get the communication working from raw flash configuration, change physical wires between ESP board and converter: RX to PIN 16, TX to PIN 17. Or change in software GPIO pins for RX and TX to 1 and 3. Boot up, run BUSSCAN (MENU 37 => 01) and you will pair board sucesfully. Tested on Supramatic E4. Cheers
